addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiAqaraSensorWeatherV1Handler.java

```diff
@@ -0,0 +1,66 @@
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Thing;
+
+import com.google.gson.JsonObject;
+
+/**
+ * Handles the Xiaomi Aqara temperature, humidity and pressure sensor
+ *
+ * @author Daniel Walters - Support for Xiaomi Aqara temperature, humidity and pressure sensor
+ */
+public class XiaomiAqaraSensorWeatherV1Handler extends XiaomiSensorBaseHandler {
```

Comment:
Same here.
Why not keep things simple? Refactor the handler for the old sensor and use it for both? Then a check, if the pressure channel exists or similar could cope the difference, what do you think?